### PR TITLE
Revert changes to drop and recreate popularity constants views

### PR DESCRIPTION
**Pull request by stacimc**
## Fixes

Fixes #[issue number] by @[issue author]

### Background

#2841 modified the popularity and data refreshes to drop and recreate the popularity constants matviews, rather than refreshing them. This was done to address the matview refresh timing out, since a similar strategy had previously worked for the media matviews.

### Problem

I neglected to realize that this is not possible for the _constants_ matviews, because regular provider DAGs now calculate popularity at ingestion and use this view to do so. Consequently, provider DAGs are failing while the view is being rebuilt.

### Fix

This PR reverts the changes to drop and recreate the popularity constants matview. Notably, this means we may start to see issues with the refresh taking a very long time again. **The popularity steps are about to be removed from the data refresh DAG, so data refreshes will not be affected. Only popularity refreshes, which are permitted to take much longer, will be affected. This will give us time to investigate other optimizations.**

Actions that must be taken:
* Wait until the current image popularity refresh DagRun passes the `create_media_popularity_constants_view` step, finishing creating the view that was dropped
* Merge this PR
* Retry all failed DAGs
* Merge changes to remove popularity steps from the data refreshes (drafted in #2818)
* Turn the image data refresh back on
* Observe the behavior of the view refresh in the popularity DAGs and investigate optimizations or extend timeouts.

## Testing Instructions

Run the data refreshes and popularity refreshes and ensure they work. Observe the DAG graphs and see that the popularity constants views are NOT dropped and then recreated, but instead refreshed.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by dhruvkb — APPROVED** [review]
Limited understanding but LGTM!

File: catalog/dags/popularity/refresh_popularity_metrics_task_factory.py [hunk 1]
```diff
@@ -68,33 +67,22 @@ def create_refresh_popularity_metrics_task_group(
...
+            task_id=UPDATE_MEDIA_POPULARITY_CONSTANTS_TASK_ID,
+            python_callable=sql.update_media_popularity_constants,
             op_kwargs={
                 "postgres_conn_id": POSTGRES_CONN_ID,
                 "media_type": media_type,
             },
             execution_timeout=execution_timeout,
             doc=(
-                "Recreates the popularity constants view. This completely "
+                "Updates the popularity constants view. This completely "
```
Comment (obulat) [inline catalog/dags/popularity/refresh_popularity_metrics_task_factory.py:79]: Is this comment accurate? Does it _completely_ recalculate the constants?

**Review by obulat — APPROVED** [review]
I ran the data refresh and popularity refresh locally, and every step was green 👍

File: catalog/dags/popularity/refresh_popularity_metrics_task_factory.py (hunk 1 above)
Comment (stacimc) [reply catalog/dags/popularity/refresh_popularity_metrics_task_factory.py:79]: It does, yes. It involves calculating the 80th percentile popularity score for each provider, from which new, updated constants are derived. This is done periodically as new data is ingested for providers.

**Comment by stacimc** [comment]
⚠️ **I am waiting to merge this PR until the constants view has been recalculated. Please do not merge.** ⚠️

**Comment by stacimc** [comment]
**UPDATE:** initially this was blocked on waiting for the view to be created -- that was because the view was being created by a task in a (running) DAG, and I did not want to push changes that would make it difficult to see the running logs through the Airflow UI. That task has since timed out and the DAG was manually failed, and the view create query is being run manually instead. It is no longer important to wait to merge this PR, so I'm going to go ahead and merge so we can be absolutely sure no DAG runs and accidentally drops it again.

**Pull request merged**